src/Factory.sol

```diff
@@ -41,7 +47,11 @@ contract Factory is Ownable, IFactory {
      * @dev Emits Deployed event
      */
     function deploy(address owner) external {
-        address proxy = address(new ERC1967Proxy(implementation, abi.encodeCall(Allocator.initialize, (owner))));
+        nonce[owner]++;
+        bytes memory initCode = abi.encodePacked(
+            type(ERC1967Proxy).creationCode, abi.encode(implementation, abi.encodeCall(Allocator.initialize, (owner)))
+        );
+        address proxy = Create2.deploy(0, keccak256(abi.encode(owner, nonce[owner])), initCode);
```

Comment:
Add tests that verify:
1. That different owners get different addresses
2. It's possible to deploy more than one contract for one owner and it gets a new address